gcc/testsuite/rust/compile/nr2/compile.exp

```diff
@@ -44,23 +44,23 @@ namespace eval rust-nr2-ns {
     # Run tests in directories
     # Manually specifying these, in case some other test file
     # does something weird
-    set test_dirs {. compile macros/builtin macros/mbe macros/proc}
+    set test_dirs {{} {macros builtin} {macros mbe} {macros proc}}
 
     set tests_expect_ok ""
     set tests_expect_err ""
 
     foreach test_dir $test_dirs {
-        foreach test [lsort [glob -nocomplain -tails -directory $srcdir/$subdir/../$test_dir *.rs]] {
-	    if {$test_dir == "."} {
-		set test_lbl $test
-	    } else {
-		set test_lbl "$test_dir/$test"
-            }
+        set directory [list {*}[file split $srcdir] {*}[file split $subdir]]
+	set directory [lreplace $directory end end]
+	set directory [list {*}$directory {*}$test_dir]
+        foreach test [lsort [glob -nocomplain -tails -directory [file join {*}$directory] *.rs]] {
+	    # use '/' as the path seperator here, for exclude file entries
```

Comment:
```suggestion
	    # use '/' as the path seperator here, to exclude file entries
```

Reply:
Ah, "exclude file entries" is the noun. I should probably quote `exclude` to make that more clear

Reply:
ah, gotcha! sorry I misread it